File: version4j/src/main/java/com/only4/algorithm/version4j/leetcode/MaxDepth.java
package com.only4.algorithm.version4j.leetcode;

import com.only4.algorithm.version4j.extra.TreeNode;

public class MaxDepth {
    public int maxDepth(TreeNode root) {
        if (root == null) return 0;

        int leftDepth = maxDepth(root.left);
        int rightDepth = maxDepth(root.right);

        return Math.max(leftDepth, rightDepth) + 1;
    }
}

File: version4k/src/main/kotlin/com/only4/algorithm/version4k/leetcode/maxDepth.kt
package com.only4.algorithm.version4k.leetcode

import com.only4.algorithm.version4k.extra.TreeNode

fun maxDepth(root: TreeNode?): Int {
    if (root == null) return 0

    val leftDepth = maxDepth(root.left)
    val rightDepth = maxDepth(root.right)

    return maxOf(leftDepth, rightDepth) + 1
}
